sklearn/cluster/_kmeans.py

```diff
@@ -1860,34 +1907,34 @@ def partial_fit(self, X, y=None, sample_weight=None):
 
             # initialize the cluster centers
             self.cluster_centers_ = self._init_centroids(
-                X, x_squared_norms=x_squared_norms,
-                init=init,
-                random_state=self._random_state,
-                init_size=self._init_size)
-
-            self._counts = np.zeros(self.n_clusters,
-                                    dtype=sample_weight.dtype)
-            random_reassign = False
-            distances = None
-        else:
-            # The lower the minimum count is, the more we do random
-            # reassignment, however, we don't want to do random
-            # reassignment too often, to allow for building up counts
-            random_reassign = self._random_state.randint(
-                10 * (1 + self._counts.min())) == 0
-            distances = np.zeros(X.shape[0], dtype=X.dtype)
-
-        _mini_batch_step(X, sample_weight, x_squared_norms,
-                         self.cluster_centers_, self._counts,
-                         np.zeros(0, dtype=X.dtype), 0,
-                         random_reassign=random_reassign, distances=distances,
-                         random_state=self._random_state,
-                         reassignment_ratio=self.reassignment_ratio,
-                         verbose=self.verbose)
+                X, x_squared_norms=x_squared_norms, init=init,
+                random_state=self._random_state, init_size=self._init_size)
+
+            # Initialize counts
+            self._counts = np.zeros(self.n_clusters, dtype=X.dtype)
+
+            # Initialize number of samples seen since last reassignment
+            self._n_since_last_reassign = 0
+
+        with threadpool_limits(limits=1, user_api="blas"):
+            _mini_batch_step(X,
+                             x_squared_norms=x_squared_norms,
+                             sample_weight=sample_weight,
+                             centers=self.cluster_centers_,
+                             centers_new=self.cluster_centers_,
+                             weight_sums=self._counts,
+                             random_state=self._random_state,
+                             random_reassign=self._random_reassign(),
+                             reassignment_ratio=self.reassignment_ratio,
+                             verbose=self.verbose,
+                             n_threads=self._n_threads)
 
         if self.compute_labels:
-            self.labels_, self.inertia_ = _labels_inertia(
-                X, sample_weight, x_squared_norms, self.cluster_centers_)
+            self.labels_, self.inertia_ = _labels_inertia_threadpool_limit(
+                X, sample_weight, x_squared_norms, self.cluster_centers_,
+                n_threads=self._n_threads)
```

Comment:
I am wondering if it makes sense to compute the labels and inertia on a minibatch. We might want to deprecate this but since it was already there, I am find with leaving this discussion outside of the scope of this PR.

Reply:
> I am find with leaving this discussion outside of the scope of this PR.

ok let's do that :)